kibana-reports/server/routes/utils/visualReportHelper.ts

```diff
@@ -18,85 +18,131 @@ import { FORMAT } from './constants';
 
 export const generatePNG = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
-    await page.setDefaultNavigationTimeout(0);
-    await page.goto(url, { waitUntil: 'networkidle0' });
-
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
-
-    // TODO: this element is for Dashboard page, need to think about addition params to select html element with source(Visualization, Dashboard)
-
-    const timeCreated = new Date().toISOString();
-    const fileName = getFileName(itemName, timeCreated) + '.' + FORMAT.png;
-
-    const buffer = await page.screenshot({
-      fullPage: true,
-    });
-
-    //TODO: Add header and footer, phase 2
-
-    await browser.close();
-    return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
+    return renderReport(
+      'image/png',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
   } catch (error) {
     throw error;
   }
 };
 
 export const generatePDF = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
+    return renderReport(
+      'application/pdf',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
+  } catch (error) {
+    throw error;
+  }
+};
 
-    await page.setDefaultNavigationTimeout(0);
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
+async function renderReport(
+  format: string,
+  url: string,
+  source: string,
+  itemName: string,
+  windowWidth: number,
+  windowHeight: number,
+  header: string,
+  footer: string
+) {
+  if (source === 'Dashboard') {
+    windowWidth = 1440;
+    windowHeight = 2560;
```

Comment:
The original is 900 * 1200, could you attach a screenshot in the PR description for review.

Reply:
Also, the width and height are from the function arguments. Don't hardcoded here

Reply:
They are function arguments but we do not provide the user with the option to configure them so they will be hard-coded in the client-side if not server-side

Reply:
Mark them as TODOs and leave comments. Fixed window size is something which will affect API and schema as well